osu.Game/Overlays/Direct/PlayButton.cs

```diff
@@ -125,6 +120,15 @@ protected override void Update()
 
         private void updatePreviewTrack(bool playing)
         {
+            if (playing && BeatmapSet == null)
```

Comment:
When is this ever the case? Why do we have random osu!direct play buttons with null beatmap sets? Why are we allowing this?

Reply:
The play button is still being displayed in the interface in the case the interface is loading.
This whole PR makes it possible to pass down a null beatmap in this case and now have the whole interface hidden, as previously.